sklearn/utils/estimator_checks.py

```diff
@@ -1720,6 +1721,47 @@ def transform(self, X):
                     shallow_params.items()))
 
 
+@ignore_warnings(category=(DeprecationWarning, FutureWarning))
+def check_set_params(name, estimator_orig):
+    if name in META_ESTIMATORS:
+        return
+    estimator = clone(estimator_orig)
+
+    # Trivial check to make sure set_params is working
+    params = estimator.get_params()
+    estimator.set_params(**params)
+
+    # Check that get_params() returns the same thing
+    # before and after set_params() with some fuzz
+    # values
+
+    estimator = clone(estimator_orig)
+    test_values = [-np.inf, np.inf, None,
```

Comment:
I don't think we need to test so many values, if we're only checking whether they are set without modification, or an error is otherwise raised. Are we trying to trigger the estimator to validate before storing??

Reply:
Yes, the idea is that if there's some sort of validation it should trigger and raise an error instead of silently trying to make a correction, e.g. clipping off negative values to 0, rounding up values, etc. Obviously if this showed up in a PR somebody could just catch it by eyeballing it but I think the whole point of the original issues was not having to eyeball it. Since there's no way in advance to know what the type of the parameter is supposed to be, I couldn't think of a better way to test this other than just trying many different kinds of extreme values, for different types.

Reply:
Shrug. I get what you're trying to do, but I'm not sure that's of enormous value, given that it lengthens the runtime of the test, etc (although maybe my impression that the overall test will take non-negligible time when run on all estimators should be tested).

The main thing we need to be testing is that `set_params` updates the `get_params` value. Testing that there is no validation is a bit of a side-point that we can only fully do by checking identity instead of equality.